networkx/algorithms/coloring/tests/test_coloring.py

```diff
@@ -22,72 +18,89 @@
     "connected_sequential",
     "saturation_largest_first",
     "DSATUR",
+    "rlf",
 ]
 
-# List of strategies where interchange=True results in an error
-INTERCHANGE_INVALID = ["independent_set", "saturation_largest_first", "DSATUR"]
-
 
 class TestColoring:
-    def test_basic_cases(self):
```

Comment:
This test suite is certainly hard to follow and the efforts to improve it are much appreciated! However, with the removal/reorganization of tests it becomes very difficult to evaluate whether the changes to the code are not breaking tested cases.

The *best* way to push this forward IMO would be to split out test suite *modifications* (i.e. changes and removals) to a separate PR that is dedicated to refactor the test suite. New tests can (and should) be *added* to this PR, but removing/changing existing tests should be left off until after the proposed changes can be shown to pass the existing test suite as is (plus any new tests).